File: Yalla_Drop/models/Order.js
const mongoose = require('mongoose');

const orderSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  restaurant: { type: mongoose.Schema.Types.ObjectId, ref: 'Restaurant', required: true },
  items: [
    {
      name: { type: String, required: true },
      quantity: { type: Number, required: true },
      price: { type: Number, required: true },
      notes: { type: String },
    },
  ],
  totalPrice: { type: Number, required: true },
  deliveryAddress: { type: String, required: true },
  deliveryNotes: { type: String },
  status: {
    type: String,
    enum: ['Pending', 'Preparing', 'Picked Up', 'On the Way', 'Delivered'],
    default: 'Pending',
  },
  tracking: {
    driverLocation: {
      coordinates: { type: [Number], index: '2dsphere' }, // Latitude & Longitude
      updatedAt: { type: Date, default: Date.now },
    },
    driverContact: { type: String }, // Driver's phone
  },
  createdAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model('Order', orderSchema);

File: app/src/main/java/com/example/yalladrop/api/Ordering/Order.kt
package com.example.yalladrop.api.Ordering

class Order (

val _id: String, // Corresponds to MongoDB _id field
val user: String, // User ID
val restaurant: String, // Restaurant ID
val items: List<OrderItem>,
val totalPrice: Double,
val deliveryAddress: String,
val deliveryNotes: String? = null,
val status: String = "Pending",
val createdAt: String

)

data class OrderItem(
    val meal: String, // Meal ID
    val quantity: Int,
    val price: Double,
    val notes: String = ""
)